### PR TITLE
Sanity check

### PR DESCRIPTION
**Pull request by jkamins7**

### CODE REVIEW

**Comment by eclee25** [comment]
@juanderone I converted the sanity check into a template but was having a couple of issues
1. Could you give some examples of some correct `npi_names` and `npi_label` assignments would be in the setup_params chunk?
2. I was getting an eval parse error in the reproductive_num chunk, perhaps related to the issue other folks were having with the empty plot in the effectiveness chunk. 
```
 plot <- temp %>%
+     filter(npi_name=="local_variance") %>%
+     mutate(county_r0 = state_r0*(1-reduction)) %>%
+     select(geoid, sim_num, county_r0, scenario, USPS, name) %>%
+     left_join(temp %>% filter(npi_name %in% npi_names)) %>%
+     mutate(r_eff = county_r0*(1-reduction)) %>%
+     group_by(!!as.symbol(geogroup), npi_name, {if(run=="state") name else(USPS)}) %>%
+     summarize(r = mean(r_eff), 
+               rq1 = quantile(r_eff, 0.25), 
+               rq3 = quantile(r_eff, 0.75),
+               reduc = mean(reduction), 
+               reducq1 = quantile(reduction, 0.25), 
+               reducq3 = quantile(reduction, 0.75)) %>%
+     rename(name = `{ ... }`)
Error in eval(parse(text = text, keep.source = FALSE), envir) : 
  '...' used in an incorrect context
```
3. I was also getting an error in the JHU_case_load chunk where case_when is trying to reassign a character vector to a logical (TRUE in this case)
```
> fig_counter <- fig_counter+1
>   jhu_CSSE_dat <- 
+     jhu_CSSE_dat %>%
+     dplyr::mutate(date = as.Date(Update)) %>%
+     dplyr::filter(Country_Region %in% country) %>%
+     dplyr::mutate(USPS = case_when(Province_State == "District of Columbia" ~ "DC", 
+                                    Province_State == "Delaware" ~ "DE", 
+                                    TRUE ~ cdlTools::fips(str_replace_all(Province_State," ", ""),to="abbreviation"))) %>%     
+     {if(run=="state") dplyr::filter(., USPS == config$sanity$state) else(.)} %>%
+     {if(run=="state") group_by(., date, FIPS, Admin2) else(group_by(., date, USPS))} %>%
+     dplyr::summarize(cumConfirmed = sum(Confirmed), cumDeaths = sum(Deaths, na.rm = TRUE)) %>%
+     ungroup() %>%
+     dplyr::arrange(date) %>%
+     {if(run=="state") dplyr::mutate(., FIPS = if_else(str_length(FIPS)==4, gsub('^', '0', FIPS), FIPS)) else(.)} %>%
+     {if(run=="state") rename(., geoid=FIPS) else(rename(.))} %>%
+     drop_na()
Error: Problem with `mutate()` input `USPS`.
x must be a character vector, not a logical vector.
 Input `USPS` is `case_when(...)`.
```

**Comment by juanderone** [comment]
> @juanderone I converted the sanity check into a template but was having a couple of issues
> 
>     1. Could you give some examples of some correct `npi_names` and `npi_label` assignments would be in the setup_params chunk? 
@eclee25 the `npi_names` should be the same as the config$interventions$settings names for the time periods were we apply inference. In the CA example, we had SchoolClosurePartial and LockdownPartial. One way to automate this would be to add "_inference" to the settings names and select `npi_names` based on that. `names(config$interventions$settings)[which(str_detect(names(config$interventions$settings), fixed("_inference", ignore_case = TRUE)))]`. 
`npi_label` is used by the plotting function for labels, so you could specify `c("School Closure (March 13-19)", "Lockdown (March 19-May 8)")`. If `npi_label` is not specified then `npi_names` are used. 

> 
>     2. I was getting an eval parse error in the reproductive_num chunk, perhaps related to the issue other folks were having with the empty plot in the effectiveness chunk.
> 
> 
> ```
>  plot <- temp %>%
> +     filter(npi_name=="local_variance") %>%
> +     mutate(county_r0 = state_r0*(1-reduction)) %>%
> +     select(geoid, sim_num, county_r0, scenario, USPS, name) %>%
> +     left_join(temp %>% filter(npi_name %in% npi_names)) %>%
> +     mutate(r_eff = county_r0*(1-reduction)) %>%
> +     group_by(!!as.symbol(geogroup), npi_name, {if(run=="state") name else(USPS)}) %>%
> +     summarize(r = mean(r_eff), 
> +               rq1 = quantile(r_eff, 0.25), 
> +               rq3 = quantile(r_eff, 0.75),
> +               reduc = mean(reduction), 
> +               reducq1 = quantile(reduction, 0.25), 
> +               reducq3 = quantile(reduction, 0.75)) %>%
> +     rename(name = `{ ... }`)
> Error in eval(parse(text = text, keep.source = FALSE), envir) : 
>   '...' used in an incorrect context
> ```
> 
This should not cause an error if `npi_names` is specified.
>     1. I was also getting an error in the JHU_case_load chunk where case_when is trying to reassign a character vector to a logical (TRUE in this case)
> 
> 
> ```
> > fig_counter <- fig_counter+1
> >   jhu_CSSE_dat <- 
> +     jhu_CSSE_dat %>%
> +     dplyr::mutate(date = as.Date(Update)) %>%
> +     dplyr::filter(Country_Region %in% country) %>%
> +     dplyr::mutate(USPS = case_when(Province_State == "District of Columbia" ~ "DC", 
> +                                    Province_State == "Delaware" ~ "DE", 
> +                                    TRUE ~ cdlTools::fips(str_replace_all(Province_State," ", ""),to="abbreviation"))) %>%     
> +     {if(run=="state") dplyr::filter(., USPS == config$sanity$state) else(.)} %>%
> +     {if(run=="state") group_by(., date, FIPS, Admin2) else(group_by(., date, USPS))} %>%
> +     dplyr::summarize(cumConfirmed = sum(Confirmed), cumDeaths = sum(Deaths, na.rm = TRUE)) %>%
> +     ungroup() %>%
> +     dplyr::arrange(date) %>%
> +     {if(run=="state") dplyr::mutate(., FIPS = if_else(str_length(FIPS)==4, gsub('^', '0', FIPS), FIPS)) else(.)} %>%
> +     {if(run=="state") rename(., geoid=FIPS) else(rename(.))} %>%
> +     drop_na()
> Error: Problem with `mutate()` input `USPS`.
> x must be a character vector, not a logical vector.
>  Input `USPS` is `case_when(...)`.
> ```
There needs to be a config$sanity$country equal to US even if it is a state run. I was thinking about the international runs... though additional changes are needed so we could have `country="US"` as default for now?

File: README.md [hunk 1]
```diff
@@ -1,21 +1,24 @@
 # COVIDScenarioPipeline
```
Comment (jkamins7) [inline README.md:1]: Wasn't expecting changes to the Readme.  What's happening here?

Comment (eclee25) [reply README.md:1]: There was a conflict in the readme that needed to be resolved. The branch originally came from master

**Comment by eclee25** [comment]
not doing this anymore

**Pull request closed without merging**